src/python/WMCore/MicroService/Unified/MSMonitor.py

```diff
@@ -0,0 +1,196 @@
+"""
+File       : MSMonitor.py
+Author     : Valentin Kuznetsov <vkuznet AT gmail dot com>
+             Alan Malta <alan dot malta AT cern dot ch >
+Description: MSMonitor class provide whole logic behind
+the transferor monitoring module.
+"""
+# futures
+from __future__ import division, print_function
+
+# system modules
+import time
+
+# WMCore modules
+from WMCore.MicroService.Unified.MSCore import MSCore
+
+
+class MSMonitor(MSCore):
+    """
+    MSMonitor class provide whole logic behind
+    the transferor monitoring module.
+    """
+    def __init__(self, msConfig, logger=None):
+        """
+        Runs the basic setup and initialization for the
+        MS Transferor module
+        :param microConfig: microservice configuration
+        """
+        super(MSMonitor, self).__init__(msConfig, logger)
+
+    def execute(self, reqStatus):
+        """
+        Executes the MS monitoring logic
+        :param reqStatus: request statue to process
+        :return:
+        """
+        try:
+            # get requests from ReqMgr2 data-service for given statue
+            # here with detail=False we get back list of records
+            requests = self.reqmgr2.getRequestByStatus([reqStatus], detail=False)
+            self.logger.debug('+++ monit found %s requests in %s state',
+                              len(requests), reqStatus)
+
+            requestStatuses = {}  # keep track of request statuses
+            for reqName in requests:
+                req = {'name': reqName, 'reqStatus': reqStatus}
+                self.logger.debug("+++ request %s", req)
+                # obtain status records from couchdb for given request
+                statusRecords = self.getStatusRecords(reqName)
+                # check status records and obtain completion status
+                statusRecords, completed = self.checkStatusRecords(statusRecords)
+                # if all transfers are completed
+                # move the request status staging -> staged
+                if completed == 100:
```

Comment:
This is 100 right now, but soon we'll have to compare against the numbers defined in the campaign configuration (and perhaps the unified config too, I don't know yet).

Reply:
Done, via this change:
https://github.com/dmwm/WMCore/pull/9276/files#diff-4cd93292a8b855fbad20e8e899adb8b7R46